### PR TITLE
[t662] - targets added several times

### PR DESCRIPTION
**Pull request by secretrobotron**
1. Fixed check for redundant target and media elements.
2. Bail when older elements are found instead of ...adding them a bunch of times.

### CODE REVIEW

**Comment by jbuck** [comment]
This needs a rebase on master. Also, are the changes in templates/test.html needed?

File: src/butter-main.js [hunk 1]
```diff
@@ -486,21 +486,28 @@
...
+
           for( i = 0, il = targets.length; i < il; ++i ) {
+            oldTarget = null;
             if( _targets.length > 0 ){
               for( j = 0, jl = _targets.length; j < jl; ++j ){
                 // don't add the same target twice
-                if( _targets[ j ].id !== targets[ i ].id ){
-                  _this.addTarget({ element: targets[ i ].id });
+                if( _targets[ j ].id === targets[ i ].id ){
+                  oldTarget = _targets[ j ];
```
Comment (jbuck) [inline src/butter-main.js:497]: Why not just make this a bool?

File: src/butter-main.js [hunk 2]
```diff
@@ -510,17 +517,20 @@
             if( _media.length > 0 ){
               for( j = 0, jl = _media.length; j < jl; ++j ){
                 if( _media[ j ].id !== medias[ i ].id && _media[ j ].url !== url ){
-                  _this.addMedia({ target: medias[ i ].id, url: url });
+                  oldMedia = _media[ j ];
```
Comment (jbuck) [inline src/butter-main.js:520]: Could make this a bool as well (=true)

File: src/butter-main.js (hunk 1 above)
Comment (secretrobotron) [inline src/butter-main.js:497]: No real reason. I was just taught to avoid using flags in uni.

File: src/butter-main.js (hunk 2 above)
Comment (secretrobotron) [inline src/butter-main.js:520]: Same reason as above.

File: templates/test.html [hunk 3]
```diff
@@ -37,7 +37,59 @@
     <video id="main" data-butter="media" controls src="../external/popcorn-js/test/trailer.ogv">
     </video>
     <div class="container-div">
-        <div class="content-div" data-butter="target" id="Area1"></div>
+        <div class="content-div" data-butter="target" id="Area1"><div></div></div>
```
Comment (jbuck) [inline templates/test.html]: Are these changes needed? It looks like the code from Brian's quiz, but we don't need that, right?
